src/main/python/treccar/trec_car_classes.py

```diff
@@ -0,0 +1,643 @@
+        pass
+    def __str__(self): return "CategoryPage"
+
+class DisambiguationPage(PageType):
+    def __init__(self):
+        pass
+    def __str__(self): return "Disambiguation Page"
+
+class RedirectPage(PageType):
+    """
+    .. attribute:: targetPage
+
+       :rtype: PageId
+
+       The target of the redirect.
+    """
+    def __init__(self, targetPage):
+        self.targetPage = targetPage
+    def __str__(self):
+        return "RedirectPage " + self.targetPage
+
+class PageMetadata(object):
+    """
+    Meta data for a page
+
+    .. attribute:: redirectNames
+
+        :rtype: PageName
+
+        Names of pages which redirect to this page
+
+    .. attribute:: disambiguationNames
+
+        :rtype: PageName
+
+        Names of disambiguation pages which link to this page
+
+    .. attribute:: disambiguationId
+
+        :rtype: PageId
+
+        Page IDs of disambiguation pages which link to this page
+
+    .. attribute:: categoryNames
+
+        :rtype: str
+
+        Page names of categories to which this page belongs
+
+    .. attribute:: categoryIds
+
+        :rtype: str
+
+        Page IDs of categories to which this page belongs
+
+    .. attribute:: inlinkIds
+
+        :rtype: str
+
+        Page IDs of pages containing inlinks
+
+    .. attribute:: inlinkAnchors
+       inlinkAnchor frequencies
+
+        :rtype: str
+
+        (Anchor text, frequency) of pages containing inlinks
+    """
+    def __init__(self, redirectNames, disambiguationNames, disambiguationIds, categoryNames, categoryIds, inlinkIds,
+                 inlinkAnchors):
+        self.inlinkAnchors = inlinkAnchors
+        self.inlinkIds = inlinkIds
+        self.categoryIds = categoryIds
+        self.categoryNames = categoryNames
+        self.disambiguationIds = disambiguationIds
+        self.disambiguationNames = disambiguationNames
+        self.redirectNames = redirectNames
+
+    @staticmethod
+    def default():
+        return PageMetadata(None, None, None, None, None, None, None)
+
+    def __str__(self):
+        redirStr = ("" if self.redirectNames is None else (" redirected = "+", ".join([name for name in self.redirectNames])))
+        disamStr = ("" if self.disambiguationNames is None else (" disambiguated = "+", ".join([name for name in self.disambiguationNames])))
+        catStr = ("" if self.redirectNames is None else (" categories = "+", ".join([name for name in self.categoryNames])))
+        inlinkStr = ("" if self.inlinkIds is None else (" inlinks = "+", ".join([name for name in self.inlinkIds])))
+        # inlinkAnchorStr = str (self.inlinkAnchors)
+        inlinkAnchorStr = ("" if self.inlinkAnchors is None else \
+                                (" inlinkAnchors = "+", ".join( \
+                                    [ ("%s: %d" % (name, freq)) for (name, freq) in self.inlinkAnchors] \
+                                    # [ ("%s: " % (name)) for (name, freq) in self.inlinkAnchors] \
+                                )))
```

Comment:
I think convention is (1) to balance each paren on its own line or (2) close paren altogether on the final line?